core/shared/src/main/scala/spire/math/Polynomial.scala

```diff
@@ -285,6 +280,59 @@ trait Polynomial[@sp(Double) C] { lhs =>
   }
 
   /**
+   * Shift this polynomial along the x-axis by `h`, so that `this(x + h) ==
+   * this.shift(h).apply(x)`.  This is equivalent to calling
+   * `this.compose(Polynomial.x + h)`, but is likely to compute the shifted
+   * polynomial much faster.
+   *
+   * @note We *could* do this without a EuclideanRing, by keeping track of the
+   * the multipliers brought down by the derivative, and cancelling as
+   * required. This will require work though, so another time.
+   */
+  def shift(h: C)(implicit ring: Ring[C], eq: Eq[C]): Polynomial[C] = {
+    // The trick here came from this answer:
+    //   http://math.stackexchange.com/questions/694565/polynomial-shift
+    // This is a heavily optimized version of the same idea. This is fairly
+    // critical method to be fast, since it is the most expensive part of the
+    // VAS root isolation algorithm.
+
+    def fromSafeLong(x: SafeLong): C =
```

Comment:
Cool - I'll leave this in here, but make a PR against algebra tonight. Once we move over to algebra, I'll just re-write this method to use it instead.